gridcapa-swe-commons/src/main/java/com/farao_community/farao/gridcapa_swe_commons/shift/SweNetworkShifter.java

```diff
@@ -151,29 +148,46 @@ private boolean isShiftSucceed(double mismatchEsPt,  double mismatchEsFr) {
         return Math.abs(mismatchEsPt) < toleranceEsPt && Math.abs(mismatchEsFr) < toleranceEsFr;
     }
 
-    private void shiftIteration(Network network, Map<String, Double> scalingValuesByCountry, ScalingParameters scalingParameters, List<String> limitingCountries, ScalableGeneratorConnector scalableGeneratorConnector) throws GlskLimitationException {
+    private Map<String, Double> shiftIteration(Network network, Map<String, Double> scalingValuesByCountry, ScalingParameters scalingParameters, ScalableGeneratorConnector scalableGeneratorConnector) throws GlskLimitationException {
+        Map<String, Double> incompleteShiftCountries = new HashMap<>();
         for (Map.Entry<String, Double> entry : scalingValuesByCountry.entrySet()) {
             String zoneId = entry.getKey();
             double asked = entry.getValue();
             String logApplyingVariationOnZone = String.format("[%s] : Applying variation on zone %s (target: %.2f)", direction, zoneId, asked);
             LOGGER.info(logApplyingVariationOnZone);
             double done = zonalScalable.getData(zoneId).scale(network, asked, scalingParameters);
             if (Math.abs(done - asked) > DEFAULT_SHIFT_EPSILON) {
-                String logWarnIncompleteVariation = String.format("[%s] : Incomplete variation on zone %s (target: %.2f, done: %.2f)",
+                String logWarnIncompleteVariation = String.format("[%s] : Incomplete shift on zone %s (target: %.2f, done: %.2f)",
                         direction, zoneId, asked, done);
                 LOGGER.warn(logWarnIncompleteVariation);
-                limitingCountries.add(zoneId);
+                incompleteShiftCountries.put(zoneId, done - asked);
             }
         }
         // During the shift some generators linked to the main network with a transformers are not connected correctly
         // Waiting for a fix in powsybl-core, we connect the transformers linked to these generators to be correctly connected to the main network component
         scalableGeneratorConnector.connectGeneratorsTransformers(network, PRE_PROCESSING_COUNTRIES);
+        return incompleteShiftCountries;
+    }
+
+    private void checkGlskLimitation(Map<String, Double> incompleteShiftCountries, double mismatchEsPt, double mismatchEsFr) throws GlskLimitationException {
+        if (incompleteShiftCountries.containsKey(toEic("PT"))) {
```

Comment:
To make it more easily readable i would declare
```
    public static final String PT = "PT";
    public static final String PT_EIC = toEic(PT);
    public static final String ES = "ES";
    public static final String ES_EIC = toEic(ES);
    public static final String FR = "FR";
    public static final String FR_EIC = toEic(FR);
```
and use these throughout the code